spec/ble-idioms.test.sh

```diff
@@ -771,3 +771,25 @@ bash: line 7: a: bad array subscript
 ## END
 ## N-I zsh/mksh/ash STDERR:
 ## END
+
+
+#### SparseArray (YSH): @[sp] and @sp
+case $SH in bash|zsh|mksh|ash) exit ;; esac
+
+a=({0..5})
+unset -v 'a[1]' 'a[2]' 'a[4]'
+var a = _a2sp(a)
+
+shopt -s parse_at
+printf '<%s>' @[a]
+echo
+printf '<%s>' @a
```

Comment:
In this test case, I want to test the result of `@[sp]`, i.e. I'd like to operate `@` on a SparseArray instance. How can I apply `${var %...}` or `["<${x %s}>" for x in ... ]` to `@[sp]`? This doesn't work:

```console
$ echo @[ [ "<${x %s}>" for x in @[a]] ]
  echo @[ [ "<${x %s}>" for x in @[a]] ]
                                 ^~
[ stdin ]:12: Unexpected token in expression mode (Id.Expr_Reserved)
```

Reply:
> I think I wanted to get rid of `printf` in YSH, and make it

Also, `printf` is a part of the POSIX shell utilities, so it is implemented as an external command even if the shell doesn't support it as a builtin command. Do you mean the ability to call external commands will be removed from YSH or the syntax for calling external commands will be different in YSH?

Reply:
Oh sorry I was just having some thoughts about the "future", not this PR

I guess the high level idea  is that we would move to something more like Python's **expressions** for string formatting, rather than a statement/command like `printf`

But yeah this PR is ready, except for the merge conflict

Reply:
I here use `argv.py`, though it is another external command and doesn't seem to be a part of the YSH language.